packages/autocomplete-core/src/getPropGetters.ts

```diff
@@ -239,19 +239,32 @@ export function getPropGetters<
     };
   };
 
-  const getLabelProps: GetLabelProps = (rest) => {
+  const getAutocompleteId = (instanceId: string, sourceId: number) => {
+    return typeof sourceId !== 'undefined'
+      ? `${instanceId}-${sourceId}`
+      : instanceId;
+  };
+
+  const getLabelProps: GetLabelProps = (providedProps) => {
+    const { sourceIndex, ...rest } = providedProps || {};
+
     return {
-      htmlFor: `${props.id}-input`,
-      id: `${props.id}-label`,
+      htmlFor: `${getAutocompleteId(props.id, sourceIndex as number)}-input`,
+      id: `${getAutocompleteId(props.id, sourceIndex as number)}-label`,
       ...rest,
     };
   };
 
-  const getListProps: GetListProps = (rest) => {
+  const getListProps: GetListProps = (providedProps) => {
+    const { sourceIndex, ...rest } = providedProps || {};
+
     return {
       role: 'listbox',
-      'aria-labelledby': `${props.id}-label`,
-      id: `${props.id}-list`,
+      'aria-labelledby': `${getAutocompleteId(
+        props.id,
+        sourceIndex as number
+      )}-label`,
+      id: `${getAutocompleteId(props.id, sourceIndex as number)}-list`,
```

Comment:
Maybe we could add `sourceIndex?: number` to `GetListProps`'s argument type so we wouldn't need to write `as number` (which is wrong too as it can be `undefined`)
That way users could also see that something will be done if `sourceIndex` is in the props

Reply:
Good point, thanks for the suggestion!